flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/batch/connectors/hive/HiveTableInputFormat.java

```diff
@@ -0,0 +1,327 @@
+ */
+public class HiveTableInputFormat extends HadoopInputFormatCommonBase<BaseRow, HiveTableInputSplit>
+		implements ResultTypeQueryable {
+	private static final long serialVersionUID = 6351448428766433164L;
+	private static Logger logger = LoggerFactory.getLogger(HiveTableInputFormat.class);
+
+	private JobConf jobConf;
+
+	protected transient Writable key;
+	protected transient Writable value;
+
+	private transient RecordReader<Writable, Writable> recordReader;
+	protected transient boolean fetched = false;
+	protected transient boolean hasNext;
+
+	private Boolean isPartitioned;
+	private BaseRowTypeInfo baseRowTypeInfo;
+
+	// Necessary info to init deserializer
+	private String[] partitionColNames;
+	private List<HiveTablePartition> partitions;
+	private transient Deserializer deserializer;
+	private transient List<? extends StructField> fieldRefs;
+	private transient StructObjectInspector oi;
+	private transient InputFormat mapredInputFormat;
+	private transient HiveTablePartition hiveTablePartition;
+
+	public HiveTableInputFormat(
+			JobConf jobConf,
+			Boolean isPartitioned,
+			String[] partitionColNames,
+			List<HiveTablePartition> partitions,
+			BaseRowTypeInfo baseRowTypeInfo) {
+		super(jobConf.getCredentials());
+		this.baseRowTypeInfo = baseRowTypeInfo;
+		this.jobConf = new JobConf(jobConf);
+		this.jobConf = jobConf;
+		this.isPartitioned = isPartitioned;
+		this.partitionColNames = partitionColNames;
+		this.partitions = partitions;
+	}
+
+	@Override
+	public void open(HiveTableInputSplit split) throws IOException {
+		this.hiveTablePartition = split.getHiveTablePartition();
+		StorageDescriptor sd = hiveTablePartition.getStorageDescriptor();
+		jobConf.set(INPUT_DIR, sd.getLocation());
+		try {
+			this.mapredInputFormat = (InputFormat)
+				Class.forName(sd.getInputFormat(), true, Thread.currentThread().getContextClassLoader()).newInstance();
+		} catch (Exception e) {
+			throw new RuntimeException("Unable to instantiate the hadoop input format", e);
+		}
+		ReflectionUtils.setConf(mapredInputFormat, jobConf);
+		if (this.mapredInputFormat instanceof Configurable) {
+			((Configurable) this.mapredInputFormat).setConf(this.jobConf);
+		} else if (this.mapredInputFormat instanceof JobConfigurable) {
+			((JobConfigurable) this.mapredInputFormat).configure(this.jobConf);
+		}
+		this.recordReader = this.mapredInputFormat.getRecordReader(split.getHadoopInputSplit(),
+			jobConf, new HadoopDummyReporter());
+		if (this.recordReader instanceof Configurable) {
+			((Configurable) this.recordReader).setConf(jobConf);
+		}
+		key = this.recordReader.createKey();
+		value = this.recordReader.createValue();
+		this.fetched = false;
+		try {
+			deserializer = (Deserializer) Class.forName(sd.getSerdeInfo().getSerializationLib()).newInstance();
+			Configuration conf = new Configuration();
+			Properties properties = HiveTableUtil.createPropertiesFromStorageDescriptor(sd);
```

Comment:
does Hive have util to create properties from SD?

Reply:
Nope.